syntax = "proto3";

package raft;

message LogEntries {
  int64 term    = 1;
  bytes command = 2;
}

message RaftUDPMessage {
    enum Type {
      HEART   = 0;
      POLL    = 1;
      VOTE    = 2;
      COMMAND = 3;
      APPEND  = 4;
      STATUS  = 5;
    }

  Type  type       = 1;
  int64 leader     = 2;
  int64 term       = 3;
  int64 sender     = 4;
  int64 commit_idx = 5;
  int64 log_idx    = 6;
  bytes command    = 7;
  repeated LogEntries log = 8;
}